compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SymbolResolver.java

```diff
@@ -258,6 +262,14 @@ private boolean hasSameOwner(BSymbol symbol, BSymbol foundSym) {
                 ((foundSym.owner.tag & SymTag.INVOKABLE) == SymTag.INVOKABLE);
     }
 
+    private boolean isSymbolRedeclaredInTestPackage(BSymbol symbol, BSymbol foundSym) {
+        if (Symbols.isFlagOn(symbol.owner.flags, Flags.TESTABLE) &&
+                !Symbols.isFlagOn(foundSym.owner.flags, Flags.TESTABLE)) {
```

Comment:
Do we need the check for the found symbol? `hasSameOwner` being false would imply it's not from the same (testable) package?

Reply:
A good suggestion. @dulvinw Can you check this and send a separate PR with this improvement.